src/aspire/denoising/denoiser_cov2d.py

```diff
@@ -18,72 +20,115 @@ class DenoiserCov2D(Denoiser):
     """
     Define a derived class for denoising 2D images using Cov2D method
     """
-    def __init__(self, src, basis, var_noise, ctf_info):
+    def __init__(self, src, basis, var_noise, ctf_info, star_file):
         """
         Initialize an object for denoising 2D images using Cov2D method
 
         :param src: The source object of 2D images with metadata
         :param basis: The basis method to expand 2D images
         :param var_noise: The estimated variance of noise
         :param ctf_info: Whether the CTF filters are included
+        :param star_file: the star file path and name to output
         """
         super().__init__(src)
-        self.imgs_noise = self.src.images(start=0, num=np.inf).asnumpy()
         self.var_noise = var_noise
         if not isinstance(basis, FFBBasis2D):
             raise NotImplementedError('Currently only fast FB method is supported')
-        self.coeffs_noise = None
-        self.coeffs_estim = None
-        self.ctf_idx = None
-        self.ctf_fb = None
         self.basis = basis
-        self.imgs_estim = None
+        self.ctf_info = ctf_info
+        self.star_file = star_file
+        self.starfile_path = os.path.dirname(star_file)
```

Comment:
Let's be consistent. It's either `star_file` or `starfile`.

Reply:
Same as above, the star file was removed from the DenoiseCov2D and used a new  `DenoisedImageSource` class for this in the new commit 23ca25c.